### PR TITLE
Maintain react syntax

### PR DESCRIPTION
**Pull request by asdacosta**
## Because

Use right language, React, for fenced code blocks

## This PR

Changes from Javascript to React syntax

## Issue

Closes #XXXXX

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: react/the_react_ecosystem/type_checking_with_proptypes.md [hunk 1]
```diff
@@ -21,15 +21,15 @@ npm install --save prop-types
 
 Next, we want to import the PropTypes package in the component whose props we want to validate.
 
-```javascript
+```jsx
```
Comment (MaoShizhong) [inline react/the_react_ecosystem/type_checking_with_proptypes.md:24]: This block doesn't need to be changed as it does not contain any JSX syntax that needs different highlighting.

File: react/the_react_ecosystem/type_checking_with_proptypes.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ export default RenderName;
 
 In this example, the component RenderName expects to receive a prop called `name` which is a string. If this prop is not a string, a warning will be displayed. If you want to make sure a prop is being passed in, use isRequired like so:
 
-```javascript
+```jsx
```
Comment (MaoShizhong) [inline react/the_react_ecosystem/type_checking_with_proptypes.md:49]: This one doesn't need changing either

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Two of the code blocks do not need changing to `jsx` - reasoning outlined in #28122

The other two code blocks do, since they actually contain JSX that needs to be properly highlighted when rendered.

**Comment by asdacosta** [comment]
> Two of the code blocks do not need changing to `jsx` - reasoning outlined in #28122
> 
> The other two code blocks do, since they actually contain JSX that needs to be properly highlighted when rendered.

Ok @MaoShizhong . Since jsx and javascript render the code the same way when no jsx is returned, should I maintain the first two I changed to jsx or I need to change it?

**Comment by MaoShizhong** [comment]
> > Two of the code blocks do not need changing to `jsx` - reasoning outlined in #28122
> > The other two code blocks do, since they actually contain JSX that needs to be properly highlighted when rendered.
> 
> Ok @MaoShizhong . Since jsx and javascript render the code the same way when no jsx is returned, should I maintain the first two I changed to jsx or I need to reverse the change?

If you could revert those two I pointed out, that would be great. We only need to set a code block's language to `jsx` if it actually contains JSX that needs JSX-specific syntax highlighting (because `javascript` won't properly do it), not just because it's a code block containing JS used in a React context (JSX isn't exclusive to React - it's just an extension of the language).

**Review by MaoShizhong — APPROVED** [review]
Many thanks, also for fixing up the knowledge check link syntax :rocket:

**Pull request merged**